Ensure the `Affected` column in Issues#index is updated when an affected node is renamed or merged

MattBudz opened the pull request:
### Summary

This PR fixes a bug where the issues table would not show an accurate list of affected nodes after an affected node has been renamed or merged.

### Check List

- [x] Added a CHANGELOG entry

MattBudz committed:
prevent multiple eventListener binds on tab change

MattBudz committed:
prevent multiple eventListener binds on tab change

MattBudz committed:
navigate to tabbed pages using browser back button

MattBudz committed:
navigate to tabbed pages using browser back button

MattBudz committed:
update changelog

MattBudz committed:
limit scope of eventLister to prevent N bindings

MattBudz committed:
limit scope of eventLister to prevent N bindings

MattBudz committed:
navigate with `Turbolinks` vs `location.reload()`

MattBudz committed:
navigate with `Turbolinks` vs `location.reload()`

MattBudz committed:
add tab to history rather than replacing current history

MattBudz committed:
add tab to history rather than replacing current history

MattBudz committed:
set data in pushState to load page with tab param

MattBudz committed:
set data in pushState to load page with tab param

MattBudz committed:
ensure issues table shows renamed affected nodes

MattBudz committed:
Update CHANGELOG

MattBudz committed:
fix rebase

aapomm commented on app/views/issues/_table.html.erb:
```diff
@@ -1,4 +1,4 @@
-<% cache ['issues-table', @all_columns, issues.map(&:id), @issues.map(&:updated_at).map(&:to_i).sort.last, @tags] do %>
+<% cache ['issues-table', @all_columns, issues.map(&:affected), issues.map(&:id), @issues.map(&:updated_at).map(&:to_i).sort.last, @tags] do %>
```
Do we need this change? If only 1 node is updated, then the whole table's cache busts. Maybe we can go with just the per-issue caching change

MattBudz replied:
Without this change, the table doesn't show the correct affected column because the cache isn't busted. Do we need to cache the table if we are caching each row?

aapomm replied:
Ah looks like I'm mistaken and the inner fragment caching are preserved and not busted: https://guides.rubyonrails.org/caching_with_rails.html#russian-doll-caching

etdsoft replied:
@aapomm why is the outer cache key  change needed then? Affected is always defined as a column.

aapomm commented on app/views/issues/_table.html.erb:
```diff
@@ -21,7 +21,7 @@
     <tbody>
       <% issues.each do |issue| %>
-        <% cache [issue, @all_columns, issue.try(:affected_count), issue.try(:state), 'issues-table', @tags] do %>
+        <% cache [issue, @all_columns, issue.try(:affected), issue.try(:state), 'issues-table', @tags] do %>
```
Instead of using all the node details in the cache key, can we instead use the `updated_at`?

```suggestion
        <% cache [issue, @all_columns, issue.try(:affected_count), issue.affected.pluck(:updated_at), issue.try(:state), 'issues-table', @tags] do %>
```

MattBudz committed:
don't store all the details of `affected` in the cache key

instead we are just keeping the updated_at value which will be enough to bust the cache

Co-authored-by: Aaron Manaloto <mail@aapomm.io>

MattBudz committed:
Merge branch 'develop' into issues/show-updated-affected-list

etdsoft reviewed (changes requested):
## Automated peer review — PR #1273

**Reviewers:** Claude (sub-agent 1), Codex (sub-agent 2, FAILED)
**Verdict:** request-changes
**Findings:** 1 High, 1 Medium, 1 Low

### What I checked

This is an automated peer code review by two independent LLM sub-agents reading the same rubric. Findings flagged by both agents are higher confidence; findings flagged by only one are still worth your attention. The verdict is computed from severity counts — there is no editorial judgment in the merge step.

### Claude only

#### F1 [High|performance] app/views/issues/_table.html.erb:24 — N+1 query via pluck in cache key

**What:** The inner cache key uses `issue.affected.pluck(:updated_at)` to track node updates:

```ruby
issue.affected.pluck(:updated_at)
```

`pluck` always issues a SQL query, even though `affected` is preloaded (via `includes(:affected)` in the controller). With N issues on the page, this adds N extra queries just to build the cache key.

**Why:** Preloaded associations are held in memory and should be accessed without additional queries. Using `pluck` defeats the preloading and causes N+1 queries on every Issues index render.

**How to fix:** Use `map` instead to use the in-memory preloaded records:

```ruby
issue.affected.map(&:updated_at)
```

If association ordering is not guaranteed, add `.sort` for key stability:

```ruby
issue.affected.map { |n| n.updated_at.to_i }.sort
```

#### F2 [Medium|hygiene] app/views/issues/_table.html.erb:PR — duplicate commits in branch history

**What:** The branch contains six commit subjects that appear twice with different hashes (e.g., "move current_project fallback", "fix nav render", etc.). These are duplicate commits from an incomplete rebase.

**Why:** Duplicate commits clutter `git log` and `git bisect` history and will appear in the merge commit. They signal a sloppy rebase workflow.

**How to fix:** Interactively rebase before merge to remove duplicates:

```bash
git rebase -i develop
```

Remove the lower (original) copies of each duplicate commit, keeping only the final version. Force-push to the branch.

#### F3 [Low|tests] app/views/issues/_table.html.erb:0 — missing test coverage for cache invalidation

**What:** No specs verify that the Issues index correctly invalidates and re-renders the Affected column when a node is renamed or merged.

**Why:** Cache invalidation is a common source of bugs, and this fix is specifically about fixing a cache-invalidation bug. Test coverage would prevent regressions.

**How to fix:** Add a request spec to `spec/requests/issues_spec.rb`:

```ruby
it "invalidates affected column cache when a node is renamed" do
  issue = create(:issue, affected: [node])
  get issues_path
  expect(response.body).to include(node.name)
  
  node.update(name: "new-name")
  get issues_path
  expect(response.body).to include("new-name")
end
```

One example is sufficient.

### Notes

**Codex sub-agent failed:** binary not found. All findings are from Claude. The N+1 query via `pluck` (F1) is the only blocker — it must be fixed before merge. Cleaning up the duplicate commits (F2) is recommended for history cleanliness but not a blocker. Test coverage (F3) is optional.

---

_Reviewed automatically. Raw outputs in `agents/product-context/reviews/dradis-ce/1273/review/`._
